noman.el

```diff
@@ -69,141 +70,115 @@
-        (noman--exec "help -m" cmd t))
-       ((string-suffix-p " not found" cmdtype)
-        (user-error "Command '%s' not found" cmdprefix))
-       (t
-        (unless (= (noman--exec cmd "--help" '(t nil)) 0)
-          (erase-buffer)
-          (noman--exec cmd "help" '(t nil))
-          (replace-regexp-in-region "." "" (point-min) (point-max)))
-        (when-let ((versioninfo
-                    (save-excursion
-                      (with-temp-buffer
-                        (unless (= (noman--exec cmdprefix "--version" '(t nil))
-                                   0)
-                          (erase-buffer)
-                          (noman--exec cmdprefix "version" '(t nil))
-                          (replace-regexp-in-region "." ""
-                                                    (point-min)
-                                                    (point-max)))
-                        (indent-code-rigidly (point-min) (point-max) 4)
-                        (buffer-string))))
-                   (versioninfo-p (not (string= (string-trim versioninfo) ""))))
-          (goto-char (point-max))
-          (insert "\nIMPLEMENTATION\n")
-          (insert versioninfo))))
+       ((string-suffix-p " is a shell builtin" type)
+        (call-process shell-file-name nil t nil "-c" (format "'%s help -m'" cmd)))
+       ((string-suffix-p " not found" type)
+        (user-error "Command '%s' not found" prefix))
+       (t (unless (= (apply #'call-process prefix nil t nil
+                            `(,@(cdr tokens) "--help"))
+                     0)
+            (erase-buffer)
+            (call-process cmd nil t nil "help")
+            (replace-regexp-in-region "." "" (point-min) (point-max)))
+          (when-let ((versioninfo
+                      (save-excursion
+                        (with-temp-buffer
+                          (unless (= (call-process prefix nil t nil  "--version") 0)
+                            (erase-buffer)
+                            (call-process prefix nil t nil "version")
+                            (replace-regexp-in-region "." "" (point-min) (point-max)))
+                          (indent-code-rigidly (point-min) (point-max) 4)
+                          (buffer-string))))
+                     (versioninfo-p (not (string-empty-p (string-trim versioninfo)))))
+            (goto-char (point-max))
+            (insert "\nIMPLEMENTATION\n")
+            (insert versioninfo))))
       (ansi-color-apply-on-region (point-min) (point-max))
-      (read-only-mode t)
-      (noman-mode)
-      (setq noman--buttons (noman--make-buttons buffer cmd))
-      (setq noman--last-command cmd)
       (goto-char (point-min))
-      (display-buffer buffer))))
+      (unless (derived-mode-p 'noman-mode) (noman-mode))
+      (setq noman--buttons (noman--buttonize cmd)
+            noman--last-command cmd
+            header-line-format (string-join (split-string cmd " ")  " > "))
```

Comment:
We can list the command/sub-commands in the header line to give the user a clear indication of what help they're viewing.